client/src/components/topics/topic-tabs.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import { ArrowDropDown, History } from "@material-ui/icons";
+import { State, TopicQuestions } from "types";
+
+import "styles/layout.css";
+import { useSelector } from "react-redux";
+import { Button, DialogActions, Paper, Tab, Tabs } from "@material-ui/core";
+import { shouldDisplayPortrait } from "pages";
+import { isMobile } from "react-device-detect";
+import { ChangeEvent } from "react";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      display: "flex",
+      flexWrap: "wrap",
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+    },
+    root: {
+      flexGrow: 1,
+    },
+  })
+);
+
+function TopicTabs(props: {
+  topicQuestions: TopicQuestions[];
+  onTopicSelected: (topic: string) => void;
+  showHistoryTab: boolean;
+}): JSX.Element {
+  const { topicQuestions, onTopicSelected, showHistoryTab } = props;
+
+  const classes = useStyles();
+  const [open, setOpen] = useState<boolean>(false);
+  const [value, setValue] = React.useState<number>(0);
```

Comment:
don't use generic variable names like `value` unless it's incredibly obvious from the context what the variable is. In this case it isn't obvious.

Is this `const [selectedTabIx, setSelectedTabIx] = React.useState<number>(0);` ?